lib/blueprinter/extractors/association_extractor.rb

```diff
@@ -5,7 +5,7 @@ def initialize
     end
 
     def extract(association_name, object, local_options, options={})
-      value = @extractor.extract(association_name, object, local_options, options)
+      value = @extractor.extract(association_name, object, local_options, options.except(:default))
```

Comment:
Why do you need to do `except(:default)`?

Reply:
I need to `except(:default)` here, because otherwise, when it makes the call through to the AutoExtractor, the AutoExtractor (with these changes) would then return the default value instead of `nil`. Then the association extractor would not pass the `return options[:default] if value.nil?` step and it would go on to call `options[:blueprint].prepare(value, view_name: view, local_options: local_options)`. You can verify that this would fail the test for default association values.

Another option (which is closer to what you are describing) we be something like this:
```
def extract(association_name, object, local_options, options={})
      value = @extractor.extract(association_name, object, local_options, options)
      if (options.key?(:default) || value.nil?)
        return value
      else
        view = options[:view] || :default
        options[:blueprint].prepare(value, view_name: view, local_options: local_options)
      end
    end
```

Thoughts?

Reply:
OHH I see ok nevermind. I think what you have is fine.